### ISSUE TITLE
No longer works with Lineage 17 (microG)

### ISSUE THREAD

**Issue by spmp** [mcastillof/FakeTraveler#34]
FakeTraveler does not work as expected in LineageOS 17.
The app itself works correctly, but I cannot _Apply_ the location.
This worked fine on Lineage 16.

**Information:**
 - OS: Lineage4microG (https://download.lineage.microg.org/tissot/)
 - Platform: Tissot
 - FakeTraveler version: latest from F-Droid and the `AndroidX` PR
 - Developer mode enabled, and `FakeTraveler` set as location mocking app.

**To Reproduce**
1. Go to main view
2. Click on `Apply`
3. The `Stop` button never appears and third party apps confirm that the location is not set.
4. See error log

**Expected behavior**
Pressing the `Apply` button would result in the button becoming `Stop` and the location being mocked.

**Logcat**
```
10-19 10:52:14.947  1483  1483 V SettingsProvider: Notifying for 0: content://settings/secure/location_providers_allowed
10-19 10:52:14.950  2900  2939 V ULocation: onSetRequest: ProviderRequest[OFF] by WorkSource{}
10-19 10:52:14.950  2900  2939 V ULocation: using autoUpdate=false autoTime=9223372036854775807
10-19 10:52:14.951  2900  2967 D ULocClient: Disable automatic updates
10-19 10:52:14.952  2969  2992 D ULocService: setUpdateInterval[com.google.android.gms] interval: 0
10-19 10:52:14.952  2969  2992 D ULocService: Disable location updates
10-19 10:52:14.952  2900  2967 D ULocClient: updateBinding - current: true, refs: 1, reqs: 0, avail: true
10-19 10:52:14.966  2900  2939 V ULocation: onSetRequest: ProviderRequest[ON interval=+1d0h0m0s0ms] by WorkSource{1000 android}
10-19 10:52:14.966  2900  2939 V ULocation: using autoUpdate=true autoTime=86400000
10-19 10:52:14.968  2900  2967 D ULocClient: Set update interval to 86400000
10-19 10:52:14.969  2969  2992 D ULocService: setUpdateInterval[com.google.android.gms] interval: 86400000
10-19 10:52:14.969  2969  2992 D ULocService: Set merged location interval to 86400000 (com.google.android.gms:86400000ms)
10-19 10:52:14.969  2900  2967 D ULocClient: updateBinding - current: true, refs: 1, reqs: 1, avail: true
```
**Additional context**
I am using the microG version of Lineage 17.

**Comment by mcastillof** [comment]
Hello @spmp. I only have a phone with LineageOS 16 and without microG or Gapps, so I can't replicate the behavior. 
The interval of seconds between every mock must be more than 0, so be sure that you don't have a 0 in the second option of the settings of the app.

If that doesn't help, there is another app in F-Droid called PrivateLocation that provides fake locations, maybe you could test if that app works for you. I'm not really sure if that app keeps pushing faked locations or it just pushes the faked location once.

**Comment by mcastillof** [comment]
Sorry... I did a quick test, and the interval seems to accept a 0, at least in LineageOS 16.

**Comment by mcastillof** [comment]
Hello. This issue is probably fixed with 4c05f6f. A new version will be released in F-Droid in a few days. I'm closing this issue, but feel free to reopen it if the problem is not solved.
